estuary-cdk/estuary_cdk/http.py

```
@@ -345,53 +346,43 @@ async def _fetch_oauth2_token(
     ) -> AccessTokenResponse:
         assert self.oauth_spec
 
-        headers = {}
-        form = {}
+        headers: dict[str, str | int] = {}
+        form: dict[str, str | int] = {"grant_type": credentials.grant_type}
+
+        match credentials.client_credentials_placement:
+            case OAuth2ClientCredentialsPlacement.HEADERS:
+                headers["Authorization"] = (
+                    "Basic "
+                    + base64.b64encode(
+                        f"{credentials.client_id}:{credentials.client_secret}".encode()
+                    ).decode()
+                )
+            case OAuth2ClientCredentialsPlacement.FORM:
+                form.update(
+                    {
+                        "client_id": credentials.client_id,
+                        "client_secret": credentials.client_secret,
+                    }
+                )
```

When doing a `match`/`case` statement that checks if a value is part of an enum, I like to include an [exhaustiveness check](https://typing.python.org/en/latest/guides/unreachable.html). This helps me catch that the `match`/`case` needs updated when a new member is added to an enum.

While we may not ever need to add a new member to the `OAuth2ClientCredentialsPlacement` enum, an exhaustiveness check here would be beneficial in case we do expand the enum later. You could use `assert_never` like mentioned in those Python docs, or you could raise an error like we do in the `match`/`case` a few lines below this point.